virttest/qemu_storage.py

```diff
@@ -92,6 +92,35 @@ def filename_to_file_opts(filename):
                 file_opts.update({'server.{i}.{k}'.format(i=i, k=k): v
                                  for i, server in enumerate(servers)
                                  for k, v in six.iteritems(server)})
+    elif re.match(r'nbd(\+\w+)?://', filename):
+        filename_pattern = re.compile(
+            r'nbd(\+(?:.+))?://((?P<host>[^/:?]+)(:(?P<port>\d+))?)?'
+            r'(/(?P<export>[^?]+))?'
+            r'(\?socket=(?P<socket>.+))?'
+        )
+        matches = filename_pattern.match(filename)
+        if matches:
+            server = {}
+            host = matches.group('host')
+            sock = matches.group('socket')
```

Comment:
How about?
```
if matches:
...
   matches = matches.groupdict()
   host = matches['host']
   port = matches['port']
   export = matches['export']
   sock = matches['sock']
....
```
since you have used `?P<xxxx>`. so you could use dict.
https://docs.python.org/3/library/re.html?highlight=re#re.Match.groupdict

Reply:
short and simple, it's better.
In fact, matches.group() was also used for other storage backends, so if we wanna a change, it's better to do code refactoring for all of them later, in a separate request. For this patch, just keep them the same as before, thanks.